stdlib/bytesLabels.mli

```diff
@@ -302,6 +302,30 @@ val equal: t -> t -> bool
 (** The equality function for byte sequences.
     @since 4.05.0 *)
 
+val to_hex : bytes -> bytes
+(** Return the given byte sequence as a byte sequence of US-ASCII lowercase
+    hexadecimal digits.
+
+    @since 4.11.0
+*)
+
+val of_hex : bytes -> bytes
+(** Convert a sequence of US-ASCII hexadecimal digits back into the
+    corresponding byte sequence.
+
+    @raise Invalid_argument if the argument is not an even number of US-ASCII
```

Comment:
Conversion functions in the Stdlib seems to raise `Invalid_argument` when checking their input by the caller is considered trivial enough (char_of_int, bool_of_string) and `Failure` when non-trivial parsing is needed (float_of_string, int_of_string).  I'd be in favor of using `Failure` here but would also be fine with dropping the raising variant and only provide the option-returning one.

Reply:
(Another) Good point, given the rename anyway... changed